File: src/test/java/com/ddd/trainings/infra/InMemoryReviewerRepo.java
package com.ddd.trainings.infra;

import com.ddd.trainings.proposal.domain.Reviewer;
import com.ddd.trainings.proposal.domain.ReviewerId;
import com.ddd.trainings.proposal.infra.ReviewerRepo;
import com.ddd.utils.InMemoryRepo;
import java.util.List;

public class InMemoryReviewerRepo
    extends InMemoryRepo<Reviewer, ReviewerId>
    implements ReviewerRepo {

  @Override
  public List<Reviewer> findByKeywordsOrderByCurrentReviewsCount(String[] value) {
    //not really accurate, but let's leave it for later
    return database.values().stream().toList();
  }
}

File: src/Ddd.Tests/Trainings/Proposal/Infra/Tests/InMemoryReviewerRepo.cs
using System.Collections.Generic;
using System.Linq;
using Ddd.Trainings.Proposal.Domain;
using Ddd.Trainings.Proposal.Infra;
using Ddd.Utils;

namespace Ddd.Tests.Trainings.Proposal.Infra.Tests;

public class InMemoryReviewerRepo : BaseRepo<Reviewer, ReviewerId>, IReviewerRepo
{
  public IEnumerable<Reviewer> FindByKeywordsOrderByCurrentReviewsCount(string[] value)
  {
    return _entities.Values.ToList();
  }
}